packages/flutter/lib/src/painting/placeholder_span.dart

```diff
@@ -15,12 +15,16 @@ import 'text_style.dart';
 /// An immutable placeholder that is embedded inline within text.
 ///
 /// [PlaceholderSpan] represents a placeholder that acts as a stand-in for other
-/// content. A [PlaceholderSpan] by itself does not contain useful
-/// information to change a [TextSpan]. Instead, this class must be extended
-/// to define contents.
+/// content. A [PlaceholderSpan] by itself does not contain useful information
+/// to change a [TextSpan]. [WidgetSpan] from the widgets library extends
+/// [PlaceholderSpan] and may be used instead to specify a widget as the contents
+/// of the placeholder.
+///
+/// Flutter widgets such as [TextField], [Text] and [RichText] do not recognize
```

Comment:
Yeah `PlaceholderSpan` doesn't know how to generate renderable content (while WidgetSpan can), but in `RenderEditable` and `RenderParagraph` we can't do `span is WidgetSpan` unfortunately. But I think we'll be able to avoid referencing `PlaceholderSpan` in this mixin, since it's using `ParentData` instead of extracting the `PlaceholderSpan`s from the `InlineSpan` tree.